engine/src/main/java/com/redhat/cloud/notifications/RunOnEngineStartup.java

```diff
@@ -1,20 +1,33 @@
 package com.redhat.cloud.notifications;
 
+import com.redhat.cloud.notifications.config.FeatureFlipper;
+import com.redhat.cloud.notifications.templates.EmailTemplateMigrationService;
+import io.quarkus.logging.Log;
 import io.quarkus.runtime.Startup;
-
 import javax.annotation.PostConstruct;
 import javax.inject.Inject;
+import java.util.List;
 
 @Startup
 public class RunOnEngineStartup {
 
     @Inject
     StartupUtils startupUtils;
 
+    @Inject
+    EmailTemplateMigrationService emailTemplateMigrationService;
+
+    @Inject
+    FeatureFlipper featureFlipper;
+
     @PostConstruct
     void postConstruct() {
         startupUtils.initAccessLogFilter();
         startupUtils.logGitProperties();
         startupUtils.logExternalServiceUrl("quarkus.rest-client.rbac-s2s.url");
+        if (featureFlipper.isUseTemplatesFromDb()) {
```

Comment:
I would feel a bit safer if we triggered the migration with a dedicated feature flag. Migrating the templates and immediately using them from the DB seems a bit risky if something goes wrong.